src/clusterfuzz/_internal/cron/schedule_fuzz.py

```diff
@@ -0,0 +1,227 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Cron job to schedule fuzz tasks that run on batch."""
+
+import collections
+import random
+import time
+from typing import Dict
+
+from googleapiclient import discovery
+
+from clusterfuzz._internal.base import concurrency
+from clusterfuzz._internal.base import tasks
+from clusterfuzz._internal.base import utils
+from clusterfuzz._internal.config import local_config
+from clusterfuzz._internal.datastore import data_types
+from clusterfuzz._internal.datastore import ndb_utils
+from clusterfuzz._internal.google_cloud_utils import credentials
+from clusterfuzz._internal.metrics import logs
+
+
+def _get_quotas(project, region):
+  gcp_credentials = credentials.get_default()[0]
+  compute = discovery.build('compute', 'v1', credentials=gcp_credentials)
+  return compute.regions().get(  # pylint: disable=no-member
+      region=region, project=project).execute()['quotas']
+
+
+def get_available_cpus(project: str, region: str) -> int:
+  """Gets the number of available CPUs in the current GCE region."""
+  quotas = _get_quotas(project, region)
+
+  # If preemptible quota is not defined, we need to use CPU quota instead.
+  cpu_quota = None
+  preemptible_quota = None
+
+  # Get preemptible_quota and cpu_quota from the list of quotas.
+  for quota in quotas:
+    if preemptible_quota and cpu_quota:
+      break
+    if quota['metric'] == 'CPUS':
+      cpu_quota = quota
+      continue
+    if quota['metric'] == 'PREEMPTIBLE_CPUS':
+      preemptible_quota = quota
+      continue
+  assert preemptible_quota or cpu_quota
+
+  if not preemptible_quota['limit']:
+    quota = cpu_quota
+  else:
+    quota = preemptible_quota
+  assert quota['limit'], quota
+
+  return quota['limit'] - quota['usage']
+
+
+class BaseFuzzTaskScheduler:
+  """Base fuzz task scheduler for any deployment of ClusterFuzz."""
+
+  def __init__(self, num_cpus):
+    self.num_cpus = num_cpus
+
+  def get_fuzz_tasks(self):
+    raise NotImplementedError('Child class must implement.')
+
+  def _get_cpus_per_fuzz_job(self, job_name):
+    del job_name
+    # TODO(metzman): Actually implement this.
+    return 2
+
+
+class FuzzerJob:
```

Comment:
this is a bit confusing. can we rename this something to differentiate it more from data_types.FuzzerJob? Something like "FuzzTaskCandidate" ?